benches/cub/device/reduce/complex.cu

```diff
@@ -0,0 +1,49 @@
+#include <nvbench/nvbench.cuh>
+
+#include "reduce_bench.cuh"
+
+template <typename DataType, unsigned int OperationsLeft>
+struct helper;
+
+template <typename DataType>
+struct helper<DataType, 1>
+{
+  static __device__ DataType compute(DataType lhs, DataType rhs)
+  {
+    return lhs + rhs;
+  }
+};
+
+template <typename DataType, unsigned int OperationsLeft>
+struct helper
+{
+  static __device__ DataType compute(DataType lhs, DataType rhs)
+  {
+    return helper<DataType, OperationsLeft - 1>::compute(sqrt(lhs * lhs),
+                                                         sqrt(rhs * rhs));
+  }
+};
+
+template <unsigned int OperationsCount>
+struct complex_op
+{
+  template <typename DataType>
+  __device__ DataType operator()(DataType lhs, DataType rhs)
+  {
+    return helper<DataType, OperationsCount>::compute(lhs, rhs);
+  }
+};
+
+using cub_custom_op = nvbench::type_list<complex_op<128>>;
+using fp_value_types = nvbench::type_list<nvbench::float32_t>;
+
+// Sweep though various input iterators and patterns using cub::Max and
```

Comment:
This comment looks unrelated. Can we rename this file to `complex_op.cu`? The name `complex` sounds like it's benchmarking reductions of complex numbers (e.g. `thrust::complex`)